src/timeout.rs

```diff
@@ -55,6 +57,29 @@ impl From<Timeout> for i32 {
     }
 }
 
+impl FromStr for Timeout {
+    type Err = ParseIntError;
+
+    /// Parse a string to a `Timeout` value.
+    ///
+    /// Takes either "default", "never" or a positive number and converts it to `Timeout`
+    /// # Example
+    /// ```
+    /// # use notify_rust::Timeout;
+    /// # use std::str::FromStr;
+    /// assert_eq!(Timeout::from_str("default").unwrap(), Timeout::Default);
+    /// assert_eq!(Timeout::from_str("never").unwrap(), Timeout::Never);
+    /// assert_eq!(Timeout::from_str("1000").unwrap(), Timeout::Milliseconds(1000));
+    /// ```
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "default" => Ok(Timeout::Default),
+            "never" => Ok(Timeout::Never),
+            milliseconds => Ok(Timeout::Milliseconds(u32::from_str(milliseconds)?)),
```

Comment:
This also allows to create `Timeout::Milliseconds(0)` by passing `"0"`, but I don't think it is worth to handle the special case, since it is also possible to just construct a `Timeout::Milliseconds(0)` directly.